internal/storage/fs/sync.go

```diff
@@ -117,15 +117,24 @@ func (s *syncedStore) CountNamespaces(ctx context.Context) (uint64, error) {
 }
 
 func (s *syncedStore) GetRollout(ctx context.Context, namespaceKey, id string) (*flipt.Rollout, error) {
-	panic("not implemented")
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+
+	return s.storeSnapshot.GetRollout(ctx, namespaceKey, id)
 }
 
 func (s *syncedStore) ListRollouts(ctx context.Context, namespaceKey, flagKey string, opts ...storage.QueryOption) (storage.ResultSet[*flipt.Rollout], error) {
-	panic("not implemented")
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+
+	return s.storeSnapshot.ListRollouts(ctx, namespaceKey, flagKey, opts...)
 }
 
 func (s *syncedStore) CountRollouts(ctx context.Context, namespaceKey, flagKey string) (uint64, error) {
-	panic("not implemented")
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+
+	return s.storeSnapshot.CountRollouts(ctx, namespaceKey, flagKey)
```

Comment:
This bit looks great, could we change all the `panic("not implemented")` in the write methods into a return with the not implemented error instead.

Same I believe goes for the snapshot implementation.